### PR TITLE
fix(sec): CVE-2024-39338 (#6539)

### PR DESCRIPTION
**Pull request by levpachmanov**
Address CVE-2024-39338 correctly following https://github.com/axios/axios/issues/6463#issuecomment-2286538061

### CODE REVIEW

**Comment by JosielHernandezDev** [comment]
I´m waiting for this fix 🥹

**Comment by jasonsaayman** [comment]
awesome thanks i was about to PR the same change

**Comment by vinialexandre** [comment]
Can someone please merge this for the love of God???

File: test/unit/regression/SNYK-JS-AXIOS-7361793.js [hunk 1]
```diff
@@ -36,10 +39,43 @@ describe('Server-Side Request Forgery (SSRF)', () => {
...
 
-        const response = await ssrfAxios.get(`/${userId}`);
-        assert.strictEqual(response.data, 'good');
-        assert.strictEqual(response.config.baseURL, 'http://localhost:' + String(GOOD_PORT));
-        assert.strictEqual(response.config.url, '//localhost:' + String(BAD_PORT));
-        assert.strictEqual(response.request.res.responseUrl, 'http://localhost:' + String(GOOD_PORT) + '/localhost:' + String(BAD_PORT));
+        try {
+            await ssrfAxios.get(`/${userId}`);
+        } catch (error) {
+            assert.strictEqual(error.message, 'Invalid URL');
```
Comment (jasonsaayman) [inline test/unit/regression/SNYK-JS-AXIOS-7361793.js:45]:
```suggestion
            assert.strictEqual(error.message, 'Invalid URL:' + userId);
```

**Comment by jasonsaayman** [comment]
> Can someone please merge this for the love of God???

Thanks, I don't think it's appropriate to cuss on the PR, this is open source I don't get paid to do this work. If you want priority on a issue i can suggest signing up to the Platinum tier on open collective

**Comment by vinialexandre** [comment]
I wasn’t cussing, just asking someone to merge it so users can have the issue resolved. But sure, carry on with your sermon, buddy.

File: test/unit/regression/SNYK-JS-AXIOS-7361793.js (hunk 1 above)
Comment (levpachmanov) [reply test/unit/regression/SNYK-JS-AXIOS-7361793.js:45]: It fails if I change `'Invalid URL:' + userId` what environment is failing on that?

**Comment by jasonsaayman** [comment]
@levpachmanov pls can you add this on ln 45 in test/unit/regression/SNYK-JS-AXIOS-7361793.js

```
assert.strictEqual(error.message, 'Invalid URL: /' + userId);
```

File: test/unit/regression/SNYK-JS-AXIOS-7361793.js (hunk 1 above)
Comment (levpachmanov) [reply test/unit/regression/SNYK-JS-AXIOS-7361793.js]: never mind, fixed that

**Comment by levpachmanov** [comment]
@jasonsaayman fixed, sorry for that

**Comment by jasonsaayman** [comment]
> @jasonsaayman fixed, sorry for that

all good 🙌

**Comment by jamesalmeida-hyper** [comment]
Could someone please explain when these changes will be released? Apologies for my inexperience with open-source projects.

**Comment by powerboat9** [comment]
> Can someone please merge this for the love of God???

https://github.com/axios/axios/blob/6b6b605eaf73852fb2dae033f1e786155959de3a/LICENSE#L7

**Comment by mate-from-mattr** [comment]
Thank you @jasonsaayman. My team at work really appreciate axios, and the work everyone's done to resolve this issue :)

**Comment by hainenber** [comment]
I almost opened another follow-up PR, that is until the merge conflict after pulled :D
Thanks @levpachmanov!

**Pull request merged**